### ISSUE TITLE
[KED 2387] Make `extra_params` and `env` accessible in ProjectHooks

### ISSUE THREAD

**Issue by marianansilva** [quantumblacklabs/kedro#666]
## Description
We are trying to define `register_config_loader ` to use a dynamic `conf_path` that changes accordingly to a variable passed as a `param` through the CLI. In order to make this possible we would like to be able to access the variable `params` passed through the CLI inside `ProjectHooks` (and `env` as well).

## Context
Consider a multi repo with configuration shared between use-cases as per bellow.

```
 conf
     artefacts/
          artefact_a/
               parameters.yml
               catalog.yml
          artefacct_b/
               parameters.yml
               catalog.yml
      use_case_1
          artefacct_a/
               .gitkeep
      use_case_2
          artefact_a/
               .gitkeep
          artefact_b/
               .gitkeep
      use_case_3/
          artefact_b/
               .gitkeep
```
We would like to build a on-deman inheritance mechanism that would allow us to share configuration from `artefacts` by adding a sub-folder inside the `use-case` config with the same name as the artefact that the use-case want to re-use. 
For that We would receive a variable called `use_case` through `kedro run --params "use_case:use_case_1"` in order to be able to define the `conf_path` accordingly and load the configuration related to the use-case together with the artefacts selected.

## Possible Implementation
(Optional) Suggest an idea for implementing the addition or change.

## Possible Alternatives
(Optional) Describe any alternative solutions or features you've considered.

**Comment by limdauto** [comment]
Hmm this is closed automatically but it will be released in 0.17.1

**Comment by idanov** [comment]
@marianansilva We have already merged the change that will solve this for you, as you can see it here aac83432f6bcb01ca875abcf7b9595f2e360c2b9. So while we are working on the next release, you can install the bleeding edge Kedro by doing: `pip install git+https://github.com/quantumblacklabs/kedro.git@master#egg=kedro`. We expect to release 0.17.1 by the end of next week, so this will be only temporary.
